### PR TITLE
Feature/add metadata type search options

### PR DESCRIPTION
**Pull request by fipa98**

### CODE REVIEW

**Comment by petterroea** [comment]
Is this required? What issue does it belong to?

**Comment by Letaguator** [comment]
#116

**Comment by Letaguator** [comment]
But the implementation is not filtering however?

**Comment by Letaguator** [comment]
/api/metadataType/by/tags
/api/metadataType/by/name

Input is supposed to be tags or name, and it should return the MetadataTypes that match these?
@fipa98 @petterroea

**Comment by petterroea** [comment]
Yes, these are supposed to return metadata matching the provided terms

**Comment by Letaguator** [comment]
Metadata or MetadataType?

**Comment by petterroea** [comment]
#116 cares about metadata types

**Comment by Letaguator** [comment]
So MetadataTypes for this PR

**Comment by petterroea** [comment]
It doesnt really make sense to search for tag and name for metadata, as the metadataType is supposed to be the handle used for categorization. The only thing differing metadata should be municipality and (soon) timespan.

Yes, correct, this PR should have endpoints for filtering metadataTypes

File: backend/Controllers/MetadataTypeController.cs [hunk 1]
```diff
@@ -141,5 +141,33 @@ public async Task<IActionResult> PostAsync([FromBody] Tag newTag, string name)
...
+		/// Get list of metadatatype tags
+		/// </summary>
+		/// <returns>
+		/// The return type of this is enum list of lists of MetadataTypeMapping
+		/// </returns>
+		[AllowAnonymous]
+		[HttpGet("by/tags")]
+		public async Task<IActionResult> GetByTagAsync()
+		{
+			var resources = await _metadataTypeService.ListTagsAsync();
```
Comment (petterroea) [inline backend/Controllers/MetadataTypeController.cs]: This isn't mapped properly. See other uses of the automapper.

The output looks like this at the moment:

```
[
  [
    {
      "tagName": "Public activity",
      "tag": null,
      "metadataTypeName": "Car History",
      "type": null
    },
    {
      "tagName": "Traffic",
      "tag": null,
      "metadataTypeName": "Car History",
      "type": null
    }
  ],
  [
    {
      "tagName": "Public activity",
      "tag": null,
      "metadataTypeName": "Cycle History",
      "type": null
    }
  ]
]
```

`tag` and `type` should not be in the response if their foreign key(`tagName` and `metadataTypeName`) are. 

Also, what is being returned isn't tags, its tag mappings

**Comment by fipa98** [comment]
No way for me to test if this works right now, because of no access to docker at the moment. Feel free to check if it works right.

**Comment by petterroea** [comment]
CI reports that the project does not build

File: backend/Controllers/MetadataTypeController.cs (hunk 1 above)
Comment (Letaguator) [reply backend/Controllers/MetadataTypeController.cs]: This should be resolved at this point.

**Comment by Letaguator** [comment]
Hmmm, weird did not notice.
Should at least build and run at this point.

**Pull request merged**